File: career/src/main/java/com/infosis/nexus/career/service/CareerPlanService.java
package com.infosis.nexus.career.service;

import com.infosis.nexus.career.dto.*;
import java.util.List;

public interface CareerPlanService {
    CareerPlanResponse create(CareerPlanRequest req);
    CareerPlanResponse getById(Long id);
    List<CareerPlanResponse> getAll();
    List<CareerPlanResponse> getByEmployee(Long employeeId);
    CareerPlanResponse update(Long id, CareerPlanRequest req);
    void delete(Long id);
    CareerPlanResponse updateProgress(Long id, Double progress);
}

File: frontend/src/services/careerPlanService.js
import api from './api';

const BASE = '/career/career-plans';

const careerPlanService = {
    getAll: () => api.get(BASE),
    getById: (id) => api.get(`${BASE}/${id}`),
    getByEmployee: (employeeId) => api.get(`${BASE}/employee/${employeeId}`),
    getRoadmap: (id) => api.get(`${BASE}/${id}/roadmap`),
    create: (data) => api.post(BASE, data),
    update: (id, data) => api.put(`${BASE}/${id}`, data),
    updateProgress: (id, progressPercentage) => api.patch(`${BASE}/${id}/progress`, { progressPercentage }),
    delete: (id) => api.delete(`${BASE}/${id}`),
};

export default careerPlanService;